### PR TITLE
[MDB IGNORE] Away Mission - Outbound Expedition

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
This PR adds a new away mission, Outbound Expedition.
This away mission is largely a departure from the current batch of away missions, having less of a focus on solely killing simplemobs, with more of a focus on exploration, puzzles, and similar content.

This away mission features:

- A large host of random events
- An objective system
- An away mission subsystem
- Almost a dozen new ruins for the away mission
- Less focus on an endless slog of simplemobs
- A variety of new machinery and UIs to play with in the away mission

[Design Document](https://hackmd.io/sMuK9jAaTZu4w555gau23Q)

image
image

Old images of (most) of the shuttle you'll be on

TL;DR
You are on a shuttle, heading towards a hitherto undefined goal, and you and your crew's goal is to make it to the end, ideally alive.

## How This Contributes To The Skyrat Roleplay Experience
We need more good gateways than Snowdin and Black Mesa

## TODO

The code here was entirely done by myself, I am aware of how much of a travesty it is, especially the later-done code. Currently, the away mission's in a "working" state that can be tested, but is absolutely not ready for merge.

- [ ] Refactor out my technical debt
- [x] Polish everything
- [x] Playtest
- [x] Rewrite Design Doc

## Changelog

:cl: Zonespace, DawsonKeyes, Paxilmaniac, DolphinE
add: New Away Mission, Outbound Expedition
/:cl:

### CODE REVIEW

**Comment by dawsonkeyes** [comment]
ill fucken map it again

**Comment by Paxilmaniac** [comment]
image

I'm not sure how you got to this point from what I did, but removing that wall on the corner of the storage area and removing everything from that middle hallway just looks fuvcked up, the reason that stuff was in the center hall to begin with was to make sure it didn't feel completely fuckin empty like it does now

**Comment by Zonespace27** [comment]
> image
> 
> I'm not sure how you got to this point from what I did, but removing that wall on the corner of the storage area and removing everything from that middle hallway just looks fuvcked up, the reason that stuff was in the center hall to begin with was to make sure it didn't feel completely fuckin empty like it does now

This has been fixed

**Comment by Zonespace27** [comment]
This is ready (enough) for undrafting

god help whoever reviews this

**Comment by Zonespace27** [comment]
This is ready (enough) for undrafting

god help whoever reviews this

File: modular_skyrat/modules/awaymissions_skyrat/code/outbound_expedition/outbound_away_controller.dm [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+			final_event_prob[event_type] = round(event_chances[event_type] / calculated_difficulty)
+		else
+			final_event_prob[event_type] = round(event_chances[event_type] * calculated_difficulty)
+	var/event_type = pick_weight(final_event_prob)
+	var/list/possible_events = event_datums.Copy()
+	for(var/datum/outbound_random_event/event as anything in possible_events)
+		if(!istype(event, event_type) || event.weight == 0)
+			possible_events.Remove(event)
+	var/pos_num = event_order.Find("random")
+	if(!is_system_dead("Sensors")) //thrusters shot? You'll only make a jump half the time
```
Comment (Zonespace27) [inline modular_skyrat/modules/awaymissions_skyrat/code/outbound_expedition/outbound_away_controller.dm]:
```suggestion
	if(!is_system_dead("Thrusters")) //thrusters shot? You'll only make a jump half the time
```

**Review by Jolly-66 — COMMENTED** [review]
I haven't looked at this PR, yet.
But going forward I'll be reviewing each map **individually** for my sanity, and to keep my reviews tidy.

**Comment by Jolly-66** [comment]
Assigning myself because I'll lose this PR locally if I hammer it with reviews.
With that being said, if @Paxilmaniac has comments they're more then welcomed to punch me in the gut.

**Comment by Zonespace27** [comment]
Hiatus for now

**Pull request closed without merging**